Amplify/Categories/DataStore/Model/Internal/Persistable.swift

```diff
@@ -67,4 +67,176 @@ struct PersistableHelper {
+        if lhs == nil && rhs == nil {
+            return false
+        }
+        switch (lhs, rhs) {
+        //case Bool Removed
+        case let (lhs, rhs) as (Temporal.Date, Temporal.Date):
+            return rhs < lhs
+        case let (lhs, rhs) as (Temporal.DateTime, Temporal.DateTime):
+            return rhs < lhs
+        case let (lhs, rhs) as (Temporal.Time, Temporal.Time):
+            return rhs < lhs
+        case let (lhs, rhs) as (Double, Double):
+            return rhs < lhs
+        case let (lhs, rhs) as (Int, Int):
+            return rhs < lhs
+        case let (lhs, rhs) as (Int, Double):
+            return rhs < Double(lhs)
+        case let (lhs, rhs) as (Double, Int):
+            return Double(rhs) < lhs
+        case let (lhs, rhs) as (String, String):
+            return rhs < lhs
+        default:
+            return false
+        }
+    }
+
+    public static func isGreaterOrEqual(_ lhs: Persistable?, _ rhs: Any?) -> Bool {
+        if lhs == nil && rhs == nil {
+            return true
+        }
+        switch (lhs, rhs) {
+        //case Bool Removed
+        case let (lhs, rhs) as (Temporal.Date, Temporal.Date):
+            return rhs >= lhs
+        case let (lhs, rhs) as (Temporal.DateTime, Temporal.DateTime):
+            return rhs >= lhs
+        case let (lhs, rhs) as (Temporal.Time, Temporal.Time):
+            return rhs >= lhs
+        case let (lhs, rhs) as (Double, Double):
+            return rhs >= lhs
+        case let (lhs, rhs) as (Int, Int):
+            return rhs >= lhs
+        case let (lhs, rhs) as (Int, Double):
+            return rhs >= Double(lhs)
+        case let (lhs, rhs) as (Double, Int):
+            return Double(rhs) >= lhs
+        case let (lhs, rhs) as (String, String):
+            return rhs >= lhs
+        default:
+            return false
+        }
+    }
+
+    public static func isGreaterThan(_ lhs: Persistable?, _ rhs: Any?) -> Bool {
+        if lhs == nil && rhs == nil {
+            return false
+        }
+        switch (lhs, rhs) {
+        //case Bool Removed
+        case let (lhs, rhs) as (Temporal.Date, Temporal.Date):
+            return rhs > lhs
+        case let (lhs, rhs) as (Temporal.DateTime, Temporal.DateTime):
+            return rhs > lhs
+        case let (lhs, rhs) as (Temporal.Time, Temporal.Time):
+            return rhs > lhs
+        case let (lhs, rhs) as (Double, Double):
+            return rhs > lhs
+        case let (lhs, rhs) as (Int, Int):
+            return rhs > lhs
+        case let (lhs, rhs) as (Double, Int):
+            return Double(rhs) > lhs
+        case let (lhs, rhs) as (Int, Double):
+            return rhs > Double(lhs)
+        case let (lhs, rhs) as (String, String):
+            return rhs > lhs
+        default:
+            return false
+        }
+    }
+
+    public static func isBetween(_ start: Persistable, _ end: Persistable, _ rhs: Any?) -> Bool {
```

Comment:
get clarity around whether or not this should include greater than or equal or just greater...

Reply:
Looking our in sql, it is suppose to be inclusive.. need to update this.

Reply:
Updated.